lib/elixir/lib/macro.ex

```diff
@@ -276,6 +276,17 @@ defmodule Macro do
     {call, line, List.insert_at([], integer, expr)}
   end
 
+  def pipe(expr, {op, line, args} = op_args, integer) when is_list(args) do
```

Comment:
I believe this is going to match first and it won't emit errors for binary operators. For example, this is not going to fail and it should: `Macro.pipe(1, quote(do: [] ++ []), 0)`

I would move the `if` inside the clause below.